### ISSUE TITLE
UDP In node stops receiving multicast data after 4 Minutes if attached to a IGMP active Switch

### ISSUE THREAD

**Issue by hammerandpick** [node-red/node-red#3062]
### What are the steps to reproduce?
Attach a device with Node-Red (Client) installed to a Switch which performs IGMP management.
Set Up a UDP Multicast Server on another (Server) and attach it to the switch, let it send multicast messages to group. (e.g. 239.12.255.254)
Create a Flow in Node-Red triggered by node "UDP IN" set Listen for to "Multicast messages" add IP4 into Group (e.g. 239.12.255.254)
Activate IGMP management on the switch.

### What happens?
Flow will perform perfectly and stop working after approx. 4 minutes.

### What do you expect to happen?
Flow to continue working without stopping

### Possible Solution
If multicast messages is selected and group has been added the client should send join request with the group after 4 minutes to update the switch's database that client is still listening and route for multicast package is sill required.
https://en.wikipedia.org/wiki/Multicast
https://en.wikipedia.org/wiki/Internet_Group_Management_Protocol

### Please tell us about your environment:
Server - Energy Meter
Client - Raspberry Pi
Switch NetGear JGS524E – 24-Port Gigabit Ethernet Plus Switch
- [ ] Node-RED version: 1.2.9 / RedMatic Version 7.2.1
- [ ] Node.js version: unknown
- [ ] npm version: unknown
- [ ] Platform/OS: Raspberry Pi/Redmatic

**Comment by dceejay** [comment]
I've never heard of this ( as presumably it's a behaviour of the switch ??? ) .
Is it a standard feature of switches - as in - is there any standard behind it ? eg for what the timeout should be ?
Or is it something that you can configure in the switch - ie not a Node-RED problem... :-)

**Comment by hammerandpick** [comment]
I had never heard of this either.
But it is correct an expected behavior according to https://datatracker.ietf.org/doc/html/rfc4541

> The snooping switch must not rely exclusively on the appearance of IGMP Group Leave announcements to determine when entries should be removed from the forwarding table.  It should implement a **membership timeout mechanism** such as the router-side functionality of the IGMP protocol as described in the IGMP and MLD specifications (See Normative Reference section for IGMPv1-3 and MLDv1-2) on all its non-router ports.  **This timeout value should be configurable.**

So Yes it can be configured in  the switch but **udp in node** should also respect that a properly configured device may dismiss a configured route after a while ;-)

**Comment by hammerandpick** [comment]
A possible solution might be if **udp in node** could receive a input signal just for triggering a multicast group join signal.

**Comment by dceejay** [comment]
Well then it would need an input pin. Just in case.  Which is a pain.  There could indeed be a rejoin tick but only if it’s a standard thing.

**Comment by hardillb** [comment]
This feels like it should be fixed in nodejs, not Node-RED, but if it is fixed it would be a check box which enables a field to adjust the timeout

**Comment by dceejay** [comment]
Agree with that

**Comment by hammerandpick** [comment]
I've raised the issue over there at node.js. https://github.com/nodejs/node/issues/39377#issue-943763082

**Comment by knolleary** [comment]
@hammerandpick keep in mind the node.js developers are unlikely to know anything about node-red specifics. Given you've described it all in terms of Node-RED nodes, they won't know what that means in terms of the node.js api.

So you're unlikely to get much response from that side unless you can point to more specific node.js apis that you think are at issue here.

**Comment by hammerandpick** [comment]
@knolleary yes, i figured that. 😢 But I am an inexperienced coder and can only try to point at the problem. And this problem is barely existing on the search engines. 
So if i describe it to the best of my knowledge someone might find a less satisfying but working solution (like turning IGMP management of) 😊

**Comment by hammerandpick** [comment]
A Similar Problem (ICMP Ping) has been solved by [coalpower/node-red-contrib-icmp-ping#node-red-contrib-icmp-ping](https://github.com/coalpower/node-red-contrib-icmp-ping#node-red-contrib-icmp-ping) .

With configuration of that node, a timeout was added, so resend a ping signal every x seconds.

Therefore my conclusion is, that it should be solved in node-red.
